core/dbt/task/deps.py

```diff
@@ -55,39 +96,149 @@ def track_package_install(
+        for package in resolved_deps:
+            lock_entry = _create_packages_yml_entry(
+                package.name, package.get_version(), package.source_type()
+            )
+            packages_installed["packages"].append(lock_entry)
+
+        with open(lock_filepath, "w") as lock_obj:
+            yaml.safe_dump(packages_installed, lock_obj)
+
+        fire_event(DepsLockCreated(lock_filepath=lock_filepath))
+
+
+class AddTask(BaseTask):
+    def __init__(self, args: Any, project: Project):
+        move_to_nearest_project_dir(project.project_root)
+        super().__init__(args=args, config=None, project=project)
+        self.cli_vars = args.vars
+
+    def check_for_duplicate_packages(self, packages_yml):
+        """Loop through contents of `packages.yml` to ensure no duplicate package names + versions.
+
+        This duplicate check will take into consideration exact match of a package name, as well as
+        a check to see if a package name exists within a name (i.e. a package name inside a git URL).
+
+        Args:
+            packages_yml (dict): In-memory read of `packages.yml` contents
+
+        Returns:
+            dict: Updated or untouched packages_yml contents
+        """
+        for i, pkg_entry in enumerate(packages_yml["packages"]):
+            for val in pkg_entry.values():
+                if self.args.package in val:
+                    del packages_yml["packages"][i]
+
+                    fire_event(DepsFoundDuplicatePackage(removed_package=pkg_entry))
+
+        return packages_yml
+
+    def run(self):
+        packages_yml_filepath = f"{self.project.project_root}/packages.yml"
+
+        if not system.path_exists(packages_yml_filepath):
+            fire_event(DepsNoPackagesFound())
+            return
+
+        if not self.args.version and self.args.source != "local":
+            fire_event(DepsVersionMissing(source=self.args.source))
+            return
+
+        new_package_entry = _create_packages_yml_entry(
+            self.args.package, self.args.version, self.args.source
+        )
+
+        with open(packages_yml_filepath, "r") as user_yml_obj:
+            packages_yml = yaml.safe_load(user_yml_obj)
+            packages_yml = self.check_for_duplicate_packages(packages_yml)
+            packages_yml["packages"].append(new_package_entry)
+
+        if packages_yml:
+            with open(packages_yml_filepath, "w") as pkg_obj:
+                yaml.safe_dump(packages_yml, pkg_obj)
+
+                fire_event(
+                    DepsAddPackage(
+                        package_name=self.args.package,
+                        version=self.args.version,
+                        packages_filepath=packages_yml_filepath,
+                    )
+                )
+
+        if not self.args.dry_run:
+            fire_event(
+                DepsLockUpdating(lock_filepath=f"{self.project.project_root}/package-lock.yml")
+            )
+            LockTask(self.args, self.project).run()
```

Comment:
Make sure I understand correctly, both `deps install`(when no lock file added before) and `deps add` will run `LockTask` and update lock yaml. Have you considered if it is not dry run `deps add` would just also install the package? I am wondering do you think it would be more natural to have `deps add` just install the newly added package. CC @dbeatty10